File: src/main/java/um/haberes/rest/model/Actividad.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import um.haberes.rest.kotlin.model.Dependencia;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "legajoId", "anho", "mes" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Actividad extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 2359306784639723089L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long actividadId;

	private Long legajoId;
	private Integer anho = 0;
	private Integer mes = 0;
	private Byte docente = 0;
	private Byte otras = 0;
	private Byte clases = 0;
	private Integer dependenciaId;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
	private Dependencia dependencia;

}

File: src/main/java/um/haberes/core/kotlin/model/Actividad.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["legajoId", "anho", "mes"])])
data class Actividad(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var actividadId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var docente: Byte = 0,
    var otras: Byte = 0,
    var clases: Byte = 0,
    var dependenciaId: Int? = null,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null

) : Auditable()
